invisible_cities/io/pmaps_io_test.py

```diff
@@ -214,14 +244,46 @@ def test_load_pmaps(output_tmpdir, data):
         write = pmpio.pmap_writer(output_file)
         list(map(write, true_pmaps, event_numbers))
 
-    read_pmaps = pmpio.load_pmaps(pmap_filename)
+        write = reio.run_and_event_writer(output_file)
+        for event_number in event_numbers:
+            write(0, event_number, 0)
+
+    read_pmaps = pmpio.load_pmaps_eager(pmap_filename)
 
     assert len(read_pmaps) == len(true_pmaps)
     assert np.all(list(read_pmaps.keys()) == event_numbers)
     for true_pmap, read_pmap in zip(true_pmaps, read_pmaps.values()):
         assert_PMap_equality(read_pmap, true_pmap)
 
 
+def test_load_pmaps_lazy(KrMC_pmaps_filename):
+    """Ensure the lazy and non-lazy versions provide the same result"""
+    pmaps_eager = pmpio.load_pmaps_eager(KrMC_pmaps_filename)
+    pmaps_lazy  = pmpio.load_pmaps_lazy (KrMC_pmaps_filename)
+
+    # combine all events in the same dictionary
+    pmaps_lazy = dict(pmaps_lazy)
+
+    assert len(pmaps_lazy) == len(pmaps_eager)
+    for evt, pmap_lazy in pmaps_lazy.items():
+        assert evt in pmaps_eager
+        assert_PMap_equality(pmap_lazy, pmaps_eager[evt])
+
+
+def test_load_pmaps_eager_lazy(KrMC_pmaps_filename):
```

Comment:
This test appears to be ensuring that the content from `load_pmaps` is as expected (type and length-wise), but the test name is a big vague in that sense. Especially since the previous test (`test_load_pmaps_lazy`) is almost the same name, but accomplishes a different task. Maybe this should be changes?

Reply:
I'm struggling to find a good name. Do you have any suggestions?

Reply:
I guess `test_load_pmaps_returns_correct_format` or `...correct_type_and_shape`? The `lazy` and `eager` are both tested so I don't think they need to be specified distinctly.